File: src/main/hava/de/richargh/sandbox/htmx/thymeleafsandbox/IndexController.java
package de.richargh.sandbox.htmx.thymeleafsandbox;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.servlet.view.RedirectView;

@Controller
public class IndexController {

    @GetMapping("/")
    public RedirectView index() {
        return new RedirectView("greeting");
    }

}

File: src/main/kotlin/de/richargh/sandbox/htmx/kotlinxhtml/IndexController.kt
package de.richargh.sandbox.htmx.kotlinxhtml

import de.richargh.sandbox.htmx.kotlinxhtml.commons.response.web.redirect
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping

@Controller
class IndexController {
    @GetMapping("/")
    fun index() = redirect("greeting")
}